pkg/kubelet/kuberuntime/logs/logs_test.go

```diff
@@ -72,137 +78,307 @@ func TestLogOptions(t *testing.T) {
 }
 
 func TestReadLogs(t *testing.T) {
-	file, err := ioutil.TempFile("", "TestFollowLogs")
+	file, err := os.CreateTemp("", "TestFollowLogs")
 	if err != nil {
 		t.Fatalf("unable to create temp file")
 	}
 	defer os.Remove(file.Name())
 	file.WriteString(`{"log":"line1\n","stream":"stdout","time":"2020-09-27T11:18:01.00000000Z"}` + "\n")
 	file.WriteString(`{"log":"line2\n","stream":"stdout","time":"2020-09-27T11:18:02.00000000Z"}` + "\n")
 	file.WriteString(`{"log":"line3\n","stream":"stdout","time":"2020-09-27T11:18:03.00000000Z"}` + "\n")
+	file.WriteString(`{"log":"line4\n","stream":"stderr","time":"2020-09-27T11:18:04.00000000Z"}` + "\n")
+	file.WriteString(`{"log":"line5\n","stream":"stdout","time":"2020-09-27T11:18:05.00000000Z"}` + "\n")
+	file.WriteString(`{"log":"line6\n","stream":"stderr","time":"2020-09-27T11:18:06.00000000Z"}` + "\n")
 
 	testCases := []struct {
-		name          string
-		podLogOptions v1.PodLogOptions
-		expected      string
+		name           string
+		podLogOptions  v1.PodLogOptions
+		expectedStdout string
+		expectedStderr string
 	}{
 		{
-			name:          "default pod log options should output all lines",
-			podLogOptions: v1.PodLogOptions{},
-			expected:      "line1\nline2\nline3\n",
+			name:           "default pod log options should output all lines",
+			podLogOptions:  v1.PodLogOptions{},
+			expectedStdout: "line1\nline2\nline3\nline5\n",
+			expectedStderr: "line4\nline6\n",
 		},
 		{
 			name: "using TailLines 2 should output last 2 lines",
 			podLogOptions: v1.PodLogOptions{
 				TailLines: pointer.Int64Ptr(2),
 			},
-			expected: "line2\nline3\n",
+			expectedStdout: "line5\n",
+			expectedStderr: "line6\n",
 		},
 		{
-			name: "using TailLines 4 should output all lines when the log has less than 4 lines",
+			name: "using TailLines 8 should output all lines when the log has less than 8 lines",
 			podLogOptions: v1.PodLogOptions{
-				TailLines: pointer.Int64Ptr(4),
+				TailLines: pointer.Int64Ptr(8),
 			},
-			expected: "line1\nline2\nline3\n",
+			expectedStdout: "line1\nline2\nline3\nline5\n",
```

Comment:
I did not expect the existing test to change - in fact, I expected this test to either add new cases, or check additional invariants, but not to change the behavior of `expected`.  

When the feature gate is off, this method should return exactly the same output in exactly the same order as it did before, and with the split here we can't prove that (and hence from review perspective I am not comfortable with the change).

I'm fine with adding `expectedStdout` and `expectedStderr` as *additional* fields, and leaving `expected` as-is (you can run ReadLogs twice if you have to in the inner loop).